checker.cs

```diff
@@ -1,38 +1,86 @@
 using System;
 using System.Diagnostics;
 
+
 class Checker
+
 {
-    static bool batteryIsOk(float temperature, float soc, float chargeRate) {
-        if(temperature < 0 || temperature > 45) {
+  
+    public static bool temperatureIsOk(float temperature)
+    {
+      if (temperature< 0 || temperature> 45)
+        {
             Console.WriteLine("Temperature is out of range!");
             return false;
-        } else if(soc < 20 || soc > 80) {
+        }
+            return true;
+    }
+
+
+    public static bool socIsOk(float soc)
+    {
+        if (soc < 20 || soc > 80)
+        {
             Console.WriteLine("State of Charge is out of range!");
             return false;
-        } else if(chargeRate > 0.8) {
+        }
+            return true;
+    }
+
+    public static bool chargeRateIsOk(float chargeRate)
+    {
+        if (chargeRate > 0.8)
+        {
             Console.WriteLine("Charge Rate is out of range!");
             return false;
         }
-        return true;
+       
+            return true;
+    }
+
+    static bool batteryIsOk(Func<float, bool> temperatureIsOk, float temperature, Func<float, bool> socIsOk, float soc, Func<float, bool> chargeRateIsOk, float chargeRate)
+    {
+
+        return temperatureIsOk(temperature);
+        return socIsOk(soc);
+        return chargeRateIsOk(chargeRate);
     }
 
-    static void ExpectTrue(bool expression) {
-        if(!expression) {
+    public static void IsExpectTrueOk(bool expression)
+    {
+        if (!expression)
+        {
             Console.WriteLine("Expected true, but got false");
             Environment.Exit(1);
```

Comment:
Generally we use Envt.Exit if there is nothing more that can be done by the code. Here , instead of exiting the program, it would be a good idea to continue test case execution.
Please see if exit is absolutely necessary...